contract/contracts/Employee.sol

```diff
@@ -0,0 +1,209 @@
+import '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/utils/PausableUpgradeable.sol';
+import '@openzeppelin/contracts/utils/structs/EnumerableSet.sol';
+
+contract Employee is OwnableUpgradeable, ReentrancyGuardUpgradeable, PausableUpgradeable {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  // Enum for offer statuses
+  enum OfferStatus {
+    Offered,
+    Accepted,
+    Rejected,
+    Resigned,
+    Fired
+  }
+
+  // Struct for salary
+  struct Salary {
+    uint256 hourlyRate;
+    uint256 hoursPerWeek;
+    uint256 signInBonus;
+    bytes8 symbol;
+  }
+
+  // Struct for employee offers with better packing
+  struct EmployeeOffer {
+    OfferStatus status;
+    uint256 startDate;
+    uint256 endDate;
+    Salary[] salary;
+    string contractUrl;
+  }
+
+  struct EmployeeOffers {
+    EmployeeOffer activeOffer;
+    EmployeeOffer pendingOffer;
+  }
+
+  // Using EnumerableSet for more efficient storage of employee addresses
+  EnumerableSet.AddressSet private employees;
+  mapping(address => EmployeeOffers) private employeeOffers;
+
+  // Events with indexed parameters for better searchability
+  event EmployeeOffered(
+    address indexed employee,
+    string contractUrl,
+    OfferStatus status
+  );
+  event EmployeeAccepted(
+    address indexed employee,
+    string contractUrl,
+    OfferStatus status
+  );
+  event EmployeeRejected(
+    address indexed employee,
+    string contractUrl,
+    OfferStatus status
+  );
+  event EmployeeResigned(
+    address indexed employee,
+    string contractUrl,
+    OfferStatus status
+  );
+  event EmployeeFired(
+    address indexed employee,
+    string contractUrl,
+    OfferStatus status
+  );
+  event EmployeeOfferApproved(
+    address indexed employee,
+    string contractUrl,
+    OfferStatus status
+  );
+
+  // Initialization function for upgradeable contract
+  function initialize() public initializer {
+    __Ownable_init(msg.sender);
+    __ReentrancyGuard_init();
+    __Pausable_init();
+  }
+
+  // Create a new offer for an employee with added checks and status management
+  function createOffer(
+    address _employee,
+    string calldata _contractUrl,
+    Salary[] calldata _salary,
+    uint256 _endDate
+  ) external onlyOwner nonReentrant whenNotPaused {
+    require(_employee != address(0), 'Invalid employee address');
+    require(bytes(_contractUrl).length > 0, 'Invalid contract URL');
+    for (uint8 i; i < _salary.length; i++) {
+      require(_salary[i].hourlyRate > 0, 'Invalid salary');
+    }
+    require(_endDate > block.timestamp, 'End date must be in the future');
+
+    // If this is a new employee, add to the employees set
+    if (!employees.contains(_employee)) {
+      employees.add(_employee);
+    }
+
+    employeeOffers[_employee].pendingOffer = EmployeeOffer({
+      contractUrl: _contractUrl,
+      salary: _salary,
+      status: OfferStatus.Offered,
+      startDate: 0,
```

Comment:
Set the start date to be the next  monday 00:00
Having the data like this will make easy the function to pay salary.